src/screens/TransactionConfirmationScreenWithData.js

```diff
@@ -46,6 +41,15 @@ class TransactionConfirmationScreenWithData extends Component {
   handleConfirmTransaction = async () => {
     const { transactionDetails } = this.props.navigation.state.params;
     const txPayload = get(transactionDetails, 'callData.params[0]');
+    let gasLimit = txPayload.gasLimit;
+    if (isNil(gasLimit)) {
+      try {
+        rawGasLimit = await web3Instance.eth.estimateGas(txPayload);
```

Comment:
This section will need to be updated given the latest common package for estimating gas with ethers